File: app/src/main/java/com/chiranths/jobportal1/Activities/LoanActivity/LoanActivity.java
package com.chiranths.jobportal1.Activities.LoanActivity;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.PagerSnapHelper;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.SnapHelper;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.Button;

import com.chiranths.jobportal1.Activities.BasicActivitys.StartingActivity;
import com.chiranths.jobportal1.Adapters.CoroselListAdaptor;
import com.chiranths.jobportal1.Adapters.LoanCoroselListAdaptor;
import com.chiranths.jobportal1.R;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import java.util.ArrayList;
import java.util.HashMap;

public class LoanActivity extends AppCompatActivity implements View.OnClickListener {

    CardView cv_PL,cv_BL,cv_HL,cv_ML,cv_LAP,cv_VL;
    Button btn_next;
    ArrayList coroselimagelist =new ArrayList();

    RecyclerView recyclerView,recyclarviewads;
    private LoanCoroselListAdaptor coroselListAdaptor;
    Handler mHandler = new Handler();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_loan);

        initilize();
    }

    private void initilize() {

      cv_PL = findViewById(R.id.cv_personal_loan);
        cv_BL = findViewById(R.id.cv_business_loan);
        cv_HL = findViewById(R.id.cv_home_loan);
        cv_ML = findViewById(R.id.cv_martgage_loan);
        cv_LAP = findViewById(R.id.cv_lap);
        cv_VL = findViewById(R.id.cv_vehicle);
        recyclerView =(RecyclerView)findViewById(R.id.rv_loan_event);
        //btn_next = findViewById(R.id.add_new_product);
        //btn_next.setOnClickListener(this);
        cv_PL.setOnClickListener(this);
        cv_BL.setOnClickListener(this);
        cv_HL.setOnClickListener(this);
        cv_LAP.setOnClickListener(this);
        cv_ML.setOnClickListener(this);
        cv_VL.setOnClickListener(this);

        AsyncTask.execute(new Runnable() {
            @Override
            public void run() {
                fetchcorosel();
            }
        });

    }

    private void fetchcorosel() {
        DatabaseReference coroselimage = FirebaseDatabase.getInstance().getReference().child("Corosels");

        coroselimage.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot snapshot) {

                if (snapshot.exists()){

                    HashMap<String, Object> dataMap = (HashMap<String, Object>) snapshot.getValue();
                    for (String key : dataMap.keySet()){
                        Object data = dataMap.get(key);
                        try{

                            HashMap<String, Object> userData = (HashMap<String, Object>) data;
                            coroselimagelist.add(userData.get("image"));

                        }catch (ClassCastException cce){

                            try{
                                String mString = String.valueOf(dataMap.get(key));
                                //addTextToView(mString);
                            }catch (ClassCastException cce2){

                            }
                        }
                    }

                    coroselListAdaptor =new LoanCoroselListAdaptor(coroselimagelist,LoanActivity.this);
                    RecyclerView.LayoutManager nlayoutManager = new LinearLayoutManager(LoanActivity.this, RecyclerView.HORIZONTAL, false);
                    recyclerView.setLayoutManager(nlayoutManager);
                    recyclerView.setItemAnimator(new DefaultItemAnimator());
                    SnapHelper snapHelper = new PagerSnapHelper();
                    snapHelper.attachToRecyclerView(recyclerView);
                    snapHelper.onFling(20,20);

                    mHandler.post(new Runnable() {
                        @Override
                        public void run() {
                            recyclerView.setAdapter(coroselListAdaptor);

                        }
                    });
                    coroselListAdaptor.notifyItemRangeInserted(0, coroselimagelist.size());

                   /* recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
                        @Override
                        public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
                            super.onScrolled(recyclerView, dx, dy);

                                mHandler2.removeCallbacks(SCROLLING_RUNNABLE);
                                Handler postHandler = new Handler();
                                postHandler.postDelayed(new Runnable() {
                                    @Override
                                    public void run() {
                                        recyclerView.setAdapter(coroselListAdaptor);
                                        recyclerView.smoothScrollBy(pixelsToMove, 0);

                                        mHandler2.postDelayed(SCROLLING_RUNNABLE, 200);
                                    }
                                }, 2000);

                        }
                    });

                    mHandler.postDelayed(SCROLLING_RUNNABLE, 200);*/
                }
            }
            @Override
            public void onCancelled(@NonNull DatabaseError error) {

            }
        });


    }

    @Override
    public void onClick(View view) {

        switch (view.getId()) {

            case R.id.cv_personal_loan:
                Intent intent = new Intent(LoanActivity.this, LoanForm.class);
                startActivity(intent);
                break;

            case R.id.cv_business_loan:
                Intent intent1 = new Intent(LoanActivity.this, LoanForm.class);
                startActivity(intent1);
                break;

            case R.id.cv_home_loan:
                Intent intent2 = new Intent(LoanActivity.this, LoanForm.class);
                startActivity(intent2);
                break;

            case R.id.cv_martgage_loan:
                Intent intent3 = new Intent(LoanActivity.this, LoanForm.class);
                startActivity(intent3);

                break;

            case R.id.cv_lap:
                Intent intent4 = new Intent(LoanActivity.this, LoanForm.class);
                startActivity(intent4);

                break;

            case R.id.cv_vehicle:
                Intent intent5 = new Intent(LoanActivity.this, LoanForm.class);
                startActivity(intent5);

                break;


        }
    }
}

File: app/src/main/java/com/chiranths/jobportal1/Activities/LoanActivity/LoanActivity.kt
package com.chiranths.jobportal1.Activities.LoanActivity

import android.content.Intent
import android.os.AsyncTask
import android.os.Bundle
import android.os.Handler
import android.view.View
import android.widget.Button
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.PagerSnapHelper
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.SnapHelper
import com.chiranths.jobportal1.Adapters.LoanCoroselListAdaptor
import com.chiranths.jobportal1.Adapters.LoanoffersAdaptor
import com.chiranths.jobportal1.Model.LoanOffersModel
import com.chiranths.jobportal1.R
import com.chiranths.jobportal1.Utilitys.AppConstants
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener

class LoanActivity : AppCompatActivity(), View.OnClickListener {
    var cv_PL: CardView? = null
    var cv_BL: CardView? = null
    var cv_HL: CardView? = null
    var cv_ML: CardView? = null
    var cv_LAP: CardView? = null
    var cv_VL: CardView? = null
    var btn_next: Button? = null
    var coroselimagelist: ArrayList<String> = ArrayList<String>()
    var bankadslist = ArrayList<LoanOffersModel>()
    var recyclerView: RecyclerView? = null
    var recyclarviewloanads: RecyclerView? = null
    var iv_nav_view : ImageView? = null
    private var coroselListAdaptor: LoanCoroselListAdaptor? = null
    private var loanoffersAdaptor: LoanoffersAdaptor? = null
    var mHandler = Handler()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_loan)
        initilize()
    }

    private fun initilize() {
        cv_PL = findViewById(R.id.cv_personal_loan)
        cv_BL = findViewById(R.id.cv_business_loan)
        cv_HL = findViewById(R.id.cv_home_loan)
        cv_ML = findViewById(R.id.cv_martgage_loan)
        cv_LAP = findViewById(R.id.cv_lap)
        cv_VL = findViewById(R.id.cv_vehicle)
        iv_nav_view = findViewById(R.id.iv_nav_view)
        recyclerView = findViewById<View>(R.id.rv_loan_event) as RecyclerView
        recyclarviewloanads = findViewById(R.id.recycler_loanoffers)
        cv_PL?.setOnClickListener(this)
        cv_BL?.setOnClickListener(this)
        cv_HL?.setOnClickListener(this)
        cv_LAP?.setOnClickListener(this)
        cv_ML?.setOnClickListener(this)
        cv_VL?.setOnClickListener(this)
        AsyncTask.execute {
            fetchcorosel()
            fetchbankads()
        }

        iv_nav_view?.setOnClickListener{
            finish()
        }
    }

    private fun fetchcorosel() {
        val coroselimage = FirebaseDatabase.getInstance().reference.child("Corosels")
        coroselimage.addValueEventListener(object : ValueEventListener {
            override fun onDataChange(snapshot: DataSnapshot) {
                if (snapshot.exists()) {
                    val dataMap = snapshot.value as HashMap<String, Any>?
                    for (key in dataMap!!.keys) {
                        val data = dataMap[key]
                        try {
                            val userData = data as HashMap<String, Any>?
                            coroselimagelist.add((userData?.get(AppConstants.image) ?: "") as String)
                        } catch (cce: ClassCastException) {

                        }
                    }
                    coroselListAdaptor = LoanCoroselListAdaptor(coroselimagelist, this@LoanActivity)
                    val nlayoutManager: RecyclerView.LayoutManager =
                        LinearLayoutManager(this@LoanActivity, RecyclerView.HORIZONTAL, false)
                    recyclerView?.layoutManager = nlayoutManager
                    recyclerView?.itemAnimator = DefaultItemAnimator()
                    val snapHelper: SnapHelper = PagerSnapHelper()
                    snapHelper.attachToRecyclerView(recyclerView)
                    snapHelper.onFling(20, 20)
                    mHandler.post { recyclerView!!.adapter = coroselListAdaptor }
                    coroselListAdaptor!!.notifyItemRangeInserted(0, coroselimagelist.size)

                }
            }

            override fun onCancelled(error: DatabaseError) {}
        })
    }

    private fun fetchbankads() {
        val coroselimage = FirebaseDatabase.getInstance().reference.child("banksadsforyou")
        coroselimage.addValueEventListener(object : ValueEventListener {
            override fun onDataChange(snapshot: DataSnapshot) {
                if (snapshot.exists()) {
                    val dataMap = snapshot.value as HashMap<String, Any>?
                    for (key in dataMap!!.keys) {
                        val data = dataMap[key]
                        try {
                            val userData = data as HashMap<String, Any>?
                            bankadslist.add(
                                LoanOffersModel(
                                    userData!![AppConstants.pid].toString(),
                                    userData["bankName"].toString(),
                                    userData["intrestrate"].toString(),
                                    userData["loanamount"].toString(),
                                    userData["loantype"].toString(),
                                    userData[AppConstants.description].toString(),
                                    userData[AppConstants.image].toString()
                                )
                            )
                        } catch (cce: ClassCastException) {

                        }
                    }
                    loanoffersAdaptor = LoanoffersAdaptor(bankadslist, this@LoanActivity)
                    val nlayoutManager: RecyclerView.LayoutManager =
                        LinearLayoutManager(this@LoanActivity, RecyclerView.VERTICAL, false)
                    recyclarviewloanads!!.layoutManager = nlayoutManager
                    recyclarviewloanads!!.itemAnimator = DefaultItemAnimator()
                    val snapHelper: SnapHelper = PagerSnapHelper()
                    snapHelper.attachToRecyclerView(recyclarviewloanads)
                    snapHelper.onFling(20, 20)
                    mHandler.post { recyclarviewloanads!!.adapter = loanoffersAdaptor }
                    loanoffersAdaptor!!.notifyItemRangeInserted(0, bankadslist.size)
                }
            }

            override fun onCancelled(error: DatabaseError) {}
        })
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.cv_personal_loan -> {
                val intent = Intent(this@LoanActivity, LoanFormfirst::class.java)
                intent.putExtra("type","1")
                startActivity(intent)
            }

            R.id.cv_business_loan -> {
                val intent1 = Intent(this@LoanActivity, LoanBusiness::class.java)
                intent.putExtra("type","2")
                startActivity(intent1)
            }

            R.id.cv_home_loan -> {
                val intent2 = Intent(this@LoanActivity, LoanFormfirst::class.java)
                intent2.putExtra("type","3")
                startActivity(intent2)
            }

            R.id.cv_martgage_loan -> {
                val intent3 = Intent(this@LoanActivity, LoanFormfirst::class.java)
                intent3.putExtra("type","4")
                startActivity(intent3)
            }

            R.id.cv_lap -> {
                val intent4 = Intent(this@LoanActivity, LoanFormfirst::class.java)
                intent4.putExtra("type","5")
                startActivity(intent4)
            }

            R.id.cv_vehicle -> {
                val intent5 = Intent(this@LoanActivity, LoanVehicle::class.java)
                intent5.putExtra("type","6")
                startActivity(intent5)
            }
        }
    }
}